File: ToDoAPI/ToDoAPI/Data/Models/ToDo.cs
﻿using System;
using System.Collections.Generic;

namespace ToDoAPI.Data.Models;

public partial class ToDo
{
    public int Id { get; set; }

    public string Description { get; set; } = null!;

    public DateTimeOffset DueDate { get; set; }

    public bool Done { get; set; }
}

File: ToDoWeb_NOTUSE/src/app/Models/ToDo.ts
export class ToDo {
    public id: number;
    public description:string;
    public dueDate : Date;
    public done: boolean;

    constructor(data: any = null){
        data = data ? JSON.parse(JSON.stringify(data)): {};

        this.id = data.id != null ? Number(data.id): null;
        this.description = data.description != null ? String(data.description): null;
        this.dueDate = data.dueDate != null ? new Date(data.dueDate): null;
        this.done = data.done != null ? Boolean(data.done): null;

    }

}